src/lib/plugin.js

```diff
@@ -29,73 +29,107 @@ module.exports = class PluginVirtual extends EventEmitter2 {
   constructor (opts) {
     super()
 
-    assertOptionType(opts, 'currencyCode', 'string')
-    assertOptionType(opts, 'currencyScale', 'number')
-    assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
-    this._currencyCode = opts.currencyCode.toUpperCase()
-    this._currencyScale = opts.currencyScale
+    this._stateful = !!opts._store
```

Comment:
Can you make this an explicit option instead and include some notes in the Readme about the ways this plugin can be run? There is quite a big difference between the two "modes" and you don't want to accidentally end up in the one that doesn't keep any records just because of a bug that caused the store not to be passed in.